src/librustdoc/doctest.rs

```diff
@@ -749,28 +769,27 @@ impl Tester for Collector {
             _ => PathBuf::from(r"doctest.rs"),
         };
 
+        // For example `module/file.rs` would become `module_file_rs`
+        let test_id = filename
+            .to_string()
+            .chars()
+            .map(|c| if c.is_ascii_alphanumeric() { c } else { '_' })
+            .collect::<String>();
+        let test_id = format!(
+            "{file}_{line}_{number}",
+            file = test_id,
```

Comment:
As mentioned in the nit comment (https://github.com/rust-lang/rust/pull/79762/files#r541196922), changing the first `let test_id =` (on line 773) to `let file =` would be more straightforward, and it would also allow you to change this line to just `file,`.

I still recommend that change.